### ISSUE TITLE
 not able to find the URL!

### ISSUE THREAD

**Issue by RezaErfani67** [Aymkdn/SharepointPlus#86]
on the first look, this syntax not work...
```javascript
$SP().list("List Name").get().then(function(data) {
  for (var i=0; i<data.length; i++) console.log(data[i].getAttribute("Title"));
});
```
Url parameter should be filled that is not good like as below
```javascript
$SP().list("List Name","MyServer").get().then(function(data) {
  for (var i=0; i<data.length; i++) console.log(data[i].getAttribute("Title"));
});
```

**Comment by Aymkdn** [comment]
Because you don't provide any details, I'll suppose you use Sharepoint 2013 On Promise. 

The URL is only required if you want to get data from another website. If you're on the same website where the list is, then this parameter is not required. You also need to be on a regular Sharepoint page (where the basics Sharepoint JS are loaded).

**Comment by RezaErfani67** [comment]
but it work correctly on sharepointplus 3 or 4...
i am using sharepoint 2010 and try to get data from current site not another...

**Comment by Aymkdn** [comment]
I always use `get` with parameters (like `fields` or `where`). Did you try with one of these parameters to see if it changes something?

I'll run some tests on next week, after Christmas break.

**Comment by RezaErfani67** [comment]
for more ensure i  will test more...
i am wating ...
thanks

**Comment by RezaErfani67** [comment]
i comment this line on sharepointplus-5.0.js, And work correctly!!  :) why?
  // if (!_this.url) throw "[SharepointPlus 'get']: not able to find the URL!";   // we cannot determine the url

**Comment by Aymkdn** [comment]
I've just tried the below code in Sharepoint 2010: 
```javascript
$SP().list("My List").get().then(function(data) { console.log(data.length) })
```

And it works perfectly.

`this.url` is setup by `list()` that will use `_getURL()` to find the URL:

1. It will first check at the cache (`_SP_BASEURL`) to see if it's already defined  
2.  If not, then it will use some variables available into the Sharepoint pages like: `L_Menu_BaseUrl` and `_spPageContextInfo.webServerRelativeUrl`  
3. If none of these solutions is defined, then it will use the web service to figure out the URL.

Please let me know the values you have into your page for : 
  * `L_Menu_BaseUrl`
  * `_spPageContextInfo.webServerRelativeUrl`

**Comment by RezaErfani67** [comment]
my function is:
```jvascript
function FillProjects2() {
    return ASQ(function(done) {
        var html = "";

        $SP().list("Project").get({
            fields: "Title,Manager,Responsible",
            where: "IsArchive <> 1",
            rowlimit: 100,
            paging: true,
            page: 1,
            orderby: "Title ASC"
        }).then(function(data){

            html += "<select class=project style=width:100%>";
            html += "<option value='' selected=true></option>";
            for (var i =data.length;i--;) {
       html+=....
            }
            html += "</select>";
            htmlProjects = html;
            console.log(data);
            done(data);

        })
    })
}
```
![1](https://user-images.githubusercontent.com/24381986/34373476-873dece8-eaf0-11e7-8bc5-899eca5150a4.jpg)
![2](https://user-images.githubusercontent.com/24381986/34373477-876c7ffe-eaf0-11e7-983f-4ce92832c51d.jpg)

**Comment by RezaErfani67** [comment]
i think  its better to be replaced as follow:

if (typeof L_Menu_BaseUrl!=="undefined")
To:
if (typeof L_Menu_BaseUrl!=="undefined" && L_Menu_BaseUrl != "" )

**Comment by Aymkdn** [comment]
I'll fix it in release 5.1. In the meantime you can define : `_SP_BASEURL="https://your.website.com/"` before using SharepointPlus.

Thanks for reporting this bug!
